readthedocs/projects/models.py

```diff
@@ -144,11 +144,19 @@ class AddonsConfig(TimeStampedModel):
     Everything is enabled by default.
     """
 
-    DOC_DIFF_DEFAULT_ROOT_SELECTOR = "[role=main]"
-    LINKPREVIEWS_DEFAULT_ROOT_SELECTOR = "[role=main] a.internal"
-    LINKPREVIEWS_DOCTOOL_NAME_CHOICES = (
-        ("sphinx", "Sphinx"),
-        ("other", "Other"),
+    DEFAULT_ROOT_SELECTOR = ", ".join(
+        [
+            "main > div > div.md-content",  # Material for MkDocs
+            "article#main",  # Docsify
+            "article",  # Pelican, Docusaurus, Antora, Jekyll
+            "div#content",  # AsciiDoctor
+            "[role=main]",
+            "main",
+            "div.body",
+            "div.document",
+            "h1",  # Very generic fallback
```

Comment:
Why do we want to target h1 before body? Wouldn't this result in the first h1 being used as the root selector -- this feels odd for diffing at least, but probably unexpected for link previews too. I guess why do we want to target h1 at all?

Reply:
Yea, in the search context, it was `h1.parent()`, which I don't know if we can specify in a CSS selector? It definitely feels to me like we'll want more logic than CSS selectors can do pretty quickly?

Reply:
You can, with `:has()`. The selector to target any element that has a direct descendant h1 is `:has(> h1)`. Direct descendant is required as `:has(h1)` will match _any_ parent element, going all the way up to `body` and `html`

Reply:
`:is()` didn't work with `selectolax` (the library we use to parse HTML on the backend) -- so, I suppose that `:has()` won't work either.

Anyways, we changed our mind here and we are deciding to move this logic to the frontend -- so, I will move this there and continue optimizing these selectors there.